src/mpi-base.jl

```diff
@@ -526,3 +527,17 @@ function ExScan{T<:MPIDatatype}(object::T, op::Op, comm::Comm)
     sendbuf = T[object]
     ExScan(sendbuf,1,op,comm)
 end
+
+# Conversion between C and Fortran Comm handles:
+
+if sizeof(CComm) == sizeof(Cint)
```

Comment:
Does this work on a 32-bit OpenMPI system? In this case, C pointers and ints will have the same size, but it may still be the case that C uses a pointer and Fortran an int, and an actual conversion is required. Changing the logic, by checking whether `MPI_Comm_f2c` exists instead of comparing type sizes, should be safer.